Sources/ViewInspector/InspectionEmissary.swift

```diff
@@ -109,54 +169,93 @@ private extension InspectionEmissary {
         return exp
     }
     
+    @available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+    func inspect<P>(onReceive publisher: P,
+                    after delay: SuspendingClock.Duration,
+                    function: String, file: StaticString, line: UInt,
+                    inspection: @escaping SubjectInspection
+    ) async throws where P: Publisher {
+        async let setup: Void = try await setup(inspection: inspection, function: function, file: file, line: line)
+        Task { @MainActor [weak notice] in
+            _ = try await publisher.values.first { _ in true }
+            let clock = SuspendingClock()
+            try await clock.sleep(until: clock.now + delay)
+            notice?.send(line)
+        }
+        try await setup
+    }
+    
     func setup(inspection: @escaping SubjectInspection,
                expectation: XCTestExpectation,
                function: String, file: StaticString, line: UInt) {
-        callbacks[line] = { [weak self] view in
-            do {
-                try inspection(view)
-            } catch let error {
-                XCTFail("\(error.localizedDescription)", file: file, line: line)
+        callbacks[line] = { view in
+            Task {
+                do {
+                    try await inspection(view)
+                } catch let error {
+                    XCTFail("\(error.localizedDescription)", file: file, line: line)
+                }
+                if await MainActor.run(body: { [weak self] in self?.callbacks.count }) == 0 {
+                    await MainActor.run { ViewHosting.expel(function: function) }
+                }
+                expectation.fulfill()
             }
-            if self?.callbacks.count == 0 {
-                ViewHosting.expel(function: function)
+        }
+    }
+    
+    @MainActor 
+    func setup(inspection: @escaping SubjectInspection,
+               function: String, file: StaticString, line: UInt) async throws {
+        try await withUnsafeThrowingContinuation { @MainActor continuation in
+            callbacks[line] = { view in
+                Task {
+                    do {
+                        continuation.resume(returning: try await inspection(view))
+                    } catch let error {
```

Comment:
```suggestion
                    } catch {
```

Reply:
I fixed this and the other inconsistent places in the codebase that do the same thing.